clients/java/src/test/java/io/camunda/client/PrivateKeyJwtIT.java

```diff
@@ -0,0 +1,112 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import dasniko.testcontainers.keycloak.KeycloakContainer;
+import io.camunda.client.impl.oauth.OAuthCredentialsProvider;
+import io.camunda.client.impl.oauth.OAuthCredentialsProviderBuilder;
+import java.io.IOException;
+import java.util.Objects;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Testcontainers
+public class PrivateKeyJwtIT {
+
+  @Container
+  static KeycloakContainer keycloak =
+      new KeycloakContainer().withRealmImportFile("privatekeyjwt/camunda-identity-test-realm.json");
+
+  private static final String KEYSTORE_PATH =
+      Objects.requireNonNull(
+              PrivateKeyJwtIT.class.getClassLoader().getResource("privatekeyjwt/keystore.p12"))
+          .getPath();
+
+  @Test
+  public void shouldReceiveAccessTokenFromIdP() throws IOException {
+    final String tokenEndpoint =
+        keycloak.getAuthServerUrl() + "/realms/camunda-identity-test/protocol/openid-connect/token";
+    // when building private key JWT credentials
+    final OAuthCredentialsProvider credentialsProvider =
+        new OAuthCredentialsProviderBuilder()
+            .authorizationServerUrl(tokenEndpoint)
+            .clientId("camunda-test")
+            .audience(tokenEndpoint)
+            .clientAssertionKeystorePath(KEYSTORE_PATH)
+            .clientAssertionKeystorePassword("password-store")
+            .clientAssertionKeystoreKeyAlias("camunda-test-1-key")
+            // WARNING some providers like Keycloak will ignore the key password as input and use
+            // only the store password to protect both the key and store itself. If two different
+            // passwords are needed, the private key and store will likely need to be generated
+            // manually. This *may* cause issues with different keystore PKCS12 implementations, and
+            // is why the generators sometimes use only one password.
+            .clientAssertionKeystoreKeyPassword("password-store")
+            .build();
+
+    // the configured IdP should recognize them as valid and return an access token
+    credentialsProvider.applyCredentials(
+        (header, value) -> {
+          assertThat(header).isEqualTo("Authorization");
+          final String token = value.replaceFirst("Bearer ", "");
+          final DecodedJWT jwt = JWT.decode(token);
+          assertThat(jwt.getType()).isEqualTo("JWT");
+        });
+  }
+
+  @Disabled("Example for MS Entra that requires a live App registration")
+  @Test
+  public void exampleEntraPrivateKeyJwtTest() throws IOException {
```

Comment:
Do you have a follow-up issue already to create Entra CI coverage (of course post-8.8 scope)? I think we should aim to have that and keep a reference there to eventually enable this test case (or move it to the appropriate place then of course). Otherwise this here remains stale code indefinitely and will eventually become outdated.